### PR TITLE
doc: Update Memfault guide to match the integration template

### PR DESCRIPTION
**Pull request by divipillai**
Update Memfault guide to match the integration template.
Minor fix to KnownIssues page.

NCDK-21319

### CODE REVIEW

**Comment by divipillai** [comment]
@umapraseeda : addressed comments suggested by you (offline). Please re-review.

File: doc/nrf/external_comp/memfault.rst [hunk 1]
```diff
@@ -110,25 +116,26 @@ Add the following code in the :file:`CMakeLists.txt` file to have these configur
 
 See the `Memfault nRF Connect SDK integration guide`_ for more details.
 
-.. _ug_memfault_error:
+.. note::
+   In the |NCS| samples and applications folders that use Memfault, these configuration files are already added.
```
Comment (umapraseeda) [inline doc/nrf/external_comp/memfault.rst:120]:
```suggestion
   In the |NCS| samples and applications that use Memfault, these configuration files are already added in the sample or application folders.
```

File: doc/nrf/external_comp/memfault.rst [hunk 2]
```diff
@@ -57,14 +57,18 @@ Integration overview
 The SDK version is included in the `west manifest file`_ in |NCS| and is automatically downloaded when running ``west update``.
 By default, it is downloaded to ``ncs/modules/lib/memfault-firmware-sdk/``.
 
-Complete the following steps to integrate Memfault into |NCS|:
+Integration Steps
```
Comment (richabp) [inline doc/nrf/external_comp/memfault.rst:60]:
```suggestion
Integration steps
```

File: doc/nrf/external_comp/memfault.rst [hunk 3]
```diff
@@ -110,25 +116,26 @@ Add the following code in the :file:`CMakeLists.txt` file to have these configur
 
 See the `Memfault nRF Connect SDK integration guide`_ for more details.
 
-.. _ug_memfault_error:
+.. note::
+   In the |NCS| samples and applications folders that use Memfault, these configuration files are already added.
 
 Error Tracking
-==============
+**************
```
Comment (richabp) [inline doc/nrf/external_comp/memfault.rst:123]: Line 122: Perhaps "Error tracking"?

File: doc/nrf/external_comp/memfault.rst [hunk 4]
```diff
@@ -153,10 +160,8 @@ The **Coredumps** section of the following sample documents provides more inform
 See `Memfault: Coredumps`_ for detailed documentation.
 Faults are captured and displayed on the **Issues** page of the Memfault UI.
 
-.. _ug_memfault_metrics:
-
 Collecting Metrics
-==================
+******************
```
Comment (richabp) [inline doc/nrf/external_comp/memfault.rst:164]: Line 163: Collecting metrics??

**Review by richabp — CHANGES_REQUESTED** [review]

File: doc/nrf/external_comp/memfault.rst (hunk 3 above)
Comment (divipillai) [reply doc/nrf/external_comp/memfault.rst:123]: In the Memfault docs, it is mentioned as "Error Tracking" - https://docs.memfault.com/docs/mcu/trace-events. So, better to keep it as is.

Comment (umapraseeda) [reply doc/nrf/external_comp/memfault.rst:123]: This needs to be checked.

File: doc/nrf/external_comp/memfault.rst (hunk 4 above)
Comment (umapraseeda) [reply doc/nrf/external_comp/memfault.rst:164]: This needs to be checked.

File: doc/nrf/external_comp/memfault.rst (hunk 3 above)
Comment (divipillai) [reply doc/nrf/external_comp/memfault.rst:123]: Checked with Jamie and Error tracking should be used. Will update the docs, wherever applicable.

File: doc/nrf/external_comp/memfault.rst (hunk 4 above)
Comment (divipillai) [reply doc/nrf/external_comp/memfault.rst:164]: Checked with Jamie and Collecting metrics should be used. Will update the docs, wherever applicable.

**Review by richabp — APPROVED** [review]

**Review by umapraseeda — APPROVED** [review]

**Pull request merged**